core/pkg/plugin/plugin_interface.go

```diff
@@ -0,0 +1,55 @@
+package plugin
+
+import (
+	"encoding/gob"
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/opencost/opencost/core/pkg/log"
+	"github.com/opencost/opencost/core/pkg/model"
+)
+
+// plugin interface
+type CustomCostSource interface {
+	GetCustomCosts(req model.CustomCostRequestInterface) []model.CustomCostResponse
+}
+
+// RPC impl
+type CustomCostRPC struct{ client *rpc.Client }
+
+func init() {
+	gob.Register(model.CustomCostRequest{})
+}
+func (c *CustomCostRPC) GetCustomCosts(req model.CustomCostRequestInterface) []model.CustomCostResponse {
```

Comment:
I include errors in the custom cost response, to allow for graceful failure (maybe some costs had no errors, but others did). Would that approach be sufficient?

Reply:
I'm a little torn; my feeling is that if `c.client.Call` returns an error, then this method should also return an error.

Reply:
so, I tried adding the error as an additional return value, however, that makes this no longer implement the interface. To avoid complicating things, I instead load the error into the existing field in the return struct. That should accomplish the goal of getting the user the actual error, while avoiding complicating the RPC interface. What do you think?

Reply:
Gah! Makes sense, that's clear!